[fix] make HTML templates more CSP compliant

return42 opened the pull request:
## What does this PR do?

Make HTML templates more CSP compliant, for details read commit messages.

## Why is this change important?

Closes: #57

## How to test this PR locally?

1. Open preference page in simple and oscar theme and have a look at the columns of the search-engines table
2. Do a `!bing_news foo` and have a look how the thumbnail is formatted.

return42 committed:
[fix] make /preferences more CSP compliant - simple & oscar theme

In simple & oscar theme remove ``style=`` properties which violates CSP but seem
not to have a magnification in GUI (UX).

Fixed files::

    ./oscar/preferences.html:25:<td class="{{ label }}" style="padding: 2px">{{- "" -}}
    ./simple/preferences.html:39:<td class="{{ label }}" style="padding: 2px; width: 13rem;">{{- "" -}}
    ./simple/preferences.html:77:    <div class="engine-tooltip" style="right: 12rem;" role="tooltip" id="{{engine_name}}_reliability">
    ./simple/preferences.html:258:     <p class="text-muted" style="margin:20px 0;">

Reported-by: https://github.com/searxng/searxng/issues/57
Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[fix] make oscar theme more CSP compliant - default-image-style

Add default-image-style in::

    searx/static/themes/oscar/src/less/result_templates.less

Replaced style= in files::

    ./oscar/result_templates/default.html:19:        <img src="{{ image_proxify(result.img_src or result.thumbnail) }}" title="{{ result.title|striptags }}" style="width: auto; max-height: 60px; min-height: 60px;" class="col-xs-2 col-sm-4 col-md-4 result-content">
    ./oscar/result_templates/files.html:24:<img src="{{ image_proxify(result.img_src) }}" alt="{{ result.title|striptags }}" title="{{ result.title|striptags }}" style="width: auto; max-height: 60px; min-height: 60px;" class="col-xs-2 col-sm-4 col-md-4 result-content">

Reported-by: https://github.com/searxng/searxng/issues/57
Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

dalf commented on searx/templates/oscar/preferences.html:
```diff
@@ -318,12 +318,12 @@ <h3>{{ _('Engines') }}</h3>
-                                    <th scope="col" style="width: 10rem">{{ _("Selected language") }}</th>
-                                    <th scope="col" style="width: 10rem">{{ _("SafeSearch") }}</th>
-                                    <th scope="col" style="width: 10rem">{{ _("Time range") }}</th>
+                                    <th scope="col">{{ _("Selected language") }}</th>
```
`width: 10rem` is useful on on wide screen: it allows to keep the column small, and give space to the other columns.

return42 replied:
Fixed in https://github.com/searxng/searxng/pull/295/commits/222031e975bc428b77a00878776744f3de884ca9

return42 committed:
[fix] make oscar theme more CSP compliant - col-checkbox

Add col-checkbox in::

    searx/static/themes/oscar/src/less/preferences.less

Replaced style in file::

    searx/templates/oscar/preferences.html

Reported-by: https://github.com/searxng/searxng/issues/57
Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

dalf commented on searx/templates/oscar/preferences.html:
```diff
@@ -318,12 +318,12 @@ <h3>{{ _('Engines') }}</h3>
-                                    <th scope="col" class="text-right"  style="width: 7rem">{{ _("Max time") }}</th>
-                                    <th scope="col" class="text-right" style="width: 7rem">{{ _("Reliablity") }}</th>
+                                    <th scope="col" class="col-checkbox text-right">{{ _("Max time") }}</th>
+                                    <th scope="col" class="col-checkbox text-right">{{ _("Reliablity") }}</th>
```
Nitpicking: col-checkbox --> col-stat
```
.col-stat {
   width: 7rem
}
```

return42 replied:
done 9068a72

dalf commented on searx/templates/oscar/result_templates/default.html:
```diff
@@ -16,7 +16,7 @@
 <div class="container-fluid">{{- "" -}}
     <div class="row">{{- "" -}}
-        <img src="{{ image_proxify(result.img_src or result.thumbnail) }}" title="{{ result.title|striptags }}" style="width: auto; max-height: 60px; min-height: 60px;" class="col-xs-2 col-sm-4 col-md-4 result-content">
+        <img src="{{ image_proxify(result.img_src or result.thumbnail) }}" title="{{ result.title|striptags }}" class="default-image-style" class="col-xs-2 col-sm-4 col-md-4 result-content">
```
There are two `class` attributes, they should be merged into one otherwise only the second one is ignored.

return42 replied:
done 5a62a8d / .. thanks for your review!

dalf commented on searx/templates/simple/preferences.html:
```diff
@@ -76,7 +76,7 @@
       </span>{{- "" -}}
     </a>{{- "" -}}
-    <div class="engine-tooltip" style="right: 12rem;" role="tooltip" id="{{engine_name}}_reliability">
+    <div class="engine-tooltip" role="tooltip" id="{{engine_name}}_reliability">
```
Without `right: 12rem` tooltip overflow outside of the screen:
![image](https://user-images.githubusercontent.com/1594191/132326460-1aac94ca-bfe3-4e27-afff-17bf0315eca0.png)

return42 replied:
done 717bf11

dalf commented on searx/templates/oscar/preferences.html:
```diff
@@ -318,12 +318,12 @@ <h3>{{ _('Engines') }}</h3>
+                                    <th scope="col" class="col-checkbox text-right">{{ _("Reliablity") }}</th>
                                     {% else %}
                                     <th scope="col">{{ _("Reliablity") }}</th>
                                     <th scope="col">{{ _("Max time") }}</th>
```
`class="col-stats"` is missing.

return42 replied:
done 9068a72

dalf reviewed (changes requested):
I left some comments.

return42 committed:
[fix] oscar theme - preferences: rename col-checkbox/col-stat

[1] https://github.com/searxng/searxng/pull/295#discussion_r703318053
[2] https://github.com/searxng/searxng/pull/295#discussion_r703337237

Suggested-by: @dalf [1] [2]
Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[fix] simple theme - preferences: fix tooltip overflow

Suggested-by: @dalf https://github.com/searxng/searxng/pull/295#discussion_r703370358
Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[fix] oscar template - merge the two class attributes into one

Suggested-by: @dalf https://github.com/searxng/searxng/pull/295#discussion_r703337237
Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

dalf commented on searx/static/themes/simple/src/less/toolkit.less:
```diff
@@ -494,6 +494,7 @@ select {
 
 /* -- engine-tooltip -- */
 .engine-tooltip {
+  right: 12rem;
```
One last point on this PR.

This CSS is for all the tooltips. With `right: 12rem`, the tooltip over the engine name will be displayed on the right too.

So a possible solution:
* `.engine-tooltip` should be remain as before
* in this line: 
  https://github.com/searxng/searxng/blob/9ef7f38e4f20bad5a76867aa3e5049a92bc58e59/searx/templates/simple/preferences.html#L73
  add a new CSS class like `column-reliability`
* in searx/static/themes/simple/src/less/preferences.less , add:
```
.column-reliability {
  .engine-tooltip {
    right: 12rem;
  }
}
```

return42 replied:
fixed in bd830f7

return42 committed:
[build] /static
